plugins/pipe_text.py

```diff
@@ -0,0 +1,59 @@
+import sublime
+import sublime_plugin
+from subprocess import run, PIPE
+import time
+from datetime import datetime
+
+
+class PipeTextCommand(sublime_plugin.TextCommand):
+    """Pipe text from ST - the selections, if any, otherwise the entire buffer contents
+       - to the specified shell command.
+       Useful for formatting XML or JSON in a quick and easy manner.
+       i.e. a workaround for https://github.com/sublimehq/sublime_text/issues/3294
+    """
+    def run(self, edit, shell_command):
+        if all((region.empty() for region in self.view.sel())):
+            regions = [sublime.Region(0, self.view.size())]
+        else:
+            regions = self.view.sel()
+
+        failures = False
+        start = time.perf_counter()
+        logs = list()
+        def log(message):
+            nonlocal logs
+            log_text = str(datetime.now()) + ' ' + message
+            logs.append(log_text)
+            print(log_text)
+
+        for region in reversed(regions):
+            text = self.view.substr(region)
+            
+            before = time.perf_counter()
+            # https://docs.python.org/3/library/subprocess.html#subprocess.run - new in version 3.5
+            # therefore, this python file should be in your User package (which defaults to Python 3.8)
+            # and you need to be using ST build >= 4050
+            p = run(shell_command, capture_output=True, input=text, encoding='utf-8')
```

Comment:
I find `shell_command` to be confusing because this call does *not* use `shell=True`.

Reply:
I agree. I consider it a bit of a "Sublime-ism" that `shell_cmd` and `cmd` carry an implication on how the task is being executed.  Is there any benefit to one over the other (or having both) for an operation such as this one?

Reply:
I'm not sure about benefits, but I've added support for both for consistency with the exec command